lib/livebook_web/live/output/terminal_text_component.ex

```diff
@@ -10,25 +10,26 @@ defmodule LivebookWeb.Output.TerminalTextComponent do
   end
 
   @impl true
-  def update(%{event: {:append, text}}, socket) do
-    {:ok, append_text(socket, text)}
+  def update(%{event: {:append, output}}, socket) do
+    {:ok, append_text(socket, output.text)}
   end
 
   def update(assigns, socket) do
-    {text, assigns} = Map.pop(assigns, :text)
+    {output, assigns} = Map.pop(assigns, :output)
     socket = assign(socket, assigns)
 
     if socket.assigns.initialized do
       {:ok, socket}
     else
       {:ok,
        socket
-       |> append_text(text)
+       |> append_text(output.text)
        |> assign(:initialized, true)}
     end
   end
 
   defp append_text(socket, text) do
+    # text = if text == :__pruned__, do: nil, else: text
     text = (socket.assigns.last_line || "") <> text
```

Comment:
This was the code that made me think that `text = if text == :__pruned__, do: nil, else: text` is never invoked, since text can be nil, and we don't handle `nil` here.